.github/workflows/e2e.yml

```diff
@@ -55,8 +55,19 @@ jobs:
         CLOVER_DATABASE_URL: postgres://${{ env.DB_USER }}:${{ env.DB_PASSWORD }}@localhost:5432/${{ env.DB_NAME }}
       run: bundle exec rake test_up
 
+    - name: Cache ngrok package
+      id: cache-ngrok
+      uses: ubicloud/cache@v4
+      with:
+        path: ngrok.tgz
+        key: ngrok-v3
+
+    - name: Download ngrok
+      if: steps.cache-ngrok.outputs.cache-hit != 'true'
+      run: curl -f -L3 https://bin.equinox.io/c/bNyj1mQVY4c/ngrok-v3-stable-linux-amd64.tgz -o ngrok.tgz
+
     - name: Install ngrok
-      run: curl -s https://ngrok-agent.s3.amazonaws.com/ngrok.asc | sudo tee /etc/apt/trusted.gpg.d/ngrok.asc >/dev/null && echo "deb https://ngrok-agent.s3.amazonaws.com buster main" | sudo tee /etc/apt/sources.list.d/ngrok.list && sudo apt update && sudo apt install ngrok
```

Comment:
This also changes the source of the binary. I see that equinox URL is used in ngrok documentation but their download page redirects to the s3 path. I feel like leaving it to apt repository is better to avoid issues in case the URL gets changes.

Reply:
Let me try to cache binary directly

Reply:
It looks like it worked https://github.com/ubicloud/ubicloud/actions/runs/10419784701/job/28858541175

Reply:
Just caching `/usr/local/bin/ngrok`